scipy/optimize/cython_optimize/__init__.py

```diff
@@ -0,0 +1,144 @@
+
+    - :py:func:`~scipy.optimize.cython_optimize.zeros_array.bisect`::
+
+            double bisect(callback_type_array f, double xa, double xb, int n, double* args, double xtol, double rtol, int iter)
+
+    - :py:func:`~scipy.optimize.cython_optimize.zeros_array.ridder`::
+
+            double ridder(callback_type_array f, double xa, double xb, int n, double* args, double xtol, double rtol, int iter)
+
+    - :py:func:`~scipy.optimize.cython_optimize.zeros_array.brenth`::
+
+            double brenth(callback_type_array f, double xa, double xb, int n, double* args, double xtol, double rtol, int iter)
+
+    - :py:func:`~scipy.optimize.cython_optimize.zeros_array.brentq`::
+
+            double brentq(callback_type_array f, double xa, double xb, int n, double* args, double xtol, double rtol, int iter)
+
+* ``scipy.optimize.cython_optimize.zeros`` exposes root finders expecting extra
+  arguments in a Python tuple.
+
+    - :py:func:`~scipy.optimize.cython_optimize.zeros.newton`::
+
+            double newton(callback_type_tuple func, double x0, callback_type_tuple fprime, tuple args, double tol, int maxiter)
+
+    - :py:func:`~scipy.optimize.cython_optimize.zeros.secant`::
+
+            double secant(callback_type_tuple func, double x0, tuple args, double tol, int maxiter)
+
+    - :py:func:`~scipy.optimize.cython_optimize.zeros.halley`::
+
+            double halley(callback_type_tuple func, double x0, callback_type_tuple fprime, tuple args, double tol, int maxiter, callback_type_tuple fprime2)
+
+    - :py:func:`~scipy.optimize.cython_optimize.zeros.bisect`::
+
+            double bisect(callback_type_tuple f, double xa, double xb, tuple args, double xtol, double rtol, int iter)
+
+    - :py:func:`~scipy.optimize.cython_optimize.zeros.ridder`::
+
+            double ridder(callback_type_tuple f, double xa, double xb, tuple args, double xtol, double rtol, int iter)
+
+    - :py:func:`~scipy.optimize.cython_optimize.zeros.brenth`::
+
+            double brenth(callback_type_tuple f, double xa, double xb, tuple args, double xtol, double rtol, int iter)
+
+    - :py:func:`~scipy.optimize.cython_optimize.zeros.brentq`::
+
+            double brentq(callback_type_tuple f, double xa, double xb, tuple args, double xtol, double rtol, int iter)
+
+Examples
+--------
+Usage of the Cython Optimize Zeros API requires the use of Cython to write
+callbacks that are compiled into native C. There are several usage examples
+with different combinations of callbacks and root-finders in the
+``cython_optimize/Examples`` folder. There are five steps:
+
+1. Select a callback signature, *eg* ``scipy.optimize.cython_optimize.zeros_struct``
+2. Select the root finder, *eg* :py:func:`~scipy.optimize.cython_optimize.zeros_struct.newton`
+3. Create a Cython ``.pyx`` file that imports the zeros module with the
```

Comment:
I cannot import any of the zeros modules in a simple, imports only file that lives *outside* the scipy directory:

```
# optimport.pyx
from scipy.optimize.cython_optimize cimport zeros
from scipy.optimize.cython_optimize cimport zeros_struct
from scipy.optimize.cython_optimize cimport zeros_array
from scipy.optimize.cython_optimize.zeros_struct cimport brentq
```

abbreviated output of ``cythonize -a -i optimport.pyx``:
```
Error compiling Cython file:
------------------------------------------------------------
...
from scipy.optimize.cython_optimize cimport zeros
from scipy.optimize.cython_optimize cimport zeros_struct
from scipy.optimize.cython_optimize cimport zeros_array
^
------------------------------------------------------------

optimport.pyx:3:0: 'scipy/optimize/cython_optimize/zeros_array.pxd' not found

Error compiling Cython file:
------------------------------------------------------------
...
from scipy.optimize.cython_optimize cimport zeros
from scipy.optimize.cython_optimize cimport zeros_struct
from scipy.optimize.cython_optimize cimport zeros_array

from scipy.optimize.cython_optimize.zeros_struct cimport brentq
^
------------------------------------------------------------

optimport.pyx:5:0: 'scipy/optimize/cython_optimize/zeros_struct/brentq.pxd' not found
Traceback (most recent call last):
  File "/Users/holmgren/miniconda3/envs/scipy2/bin/cythonize", line 11, in <module>
    sys.exit(main())
  File "/Users/holmgren/miniconda3/envs/scipy2/lib/python3.6/site-packages/Cython/Build/Cythonize.py", line 196, in main
    cython_compile(path, options)
  File "/Users/holmgren/miniconda3/envs/scipy2/lib/python3.6/site-packages/Cython/Build/Cythonize.py", line 90, in cython_compile
    **options.options)
  File "/Users/holmgren/miniconda3/envs/scipy2/lib/python3.6/site-packages/Cython/Build/Dependencies.py", line 1026, in cythonize
    cythonize_one(*args)
  File "/Users/holmgren/miniconda3/envs/scipy2/lib/python3.6/site-packages/Cython/Build/Dependencies.py", line 1146, in cythonize_one
    raise CompileError(None, pyx_file)
Cython.Compiler.Errors.CompileError: /Users/holmgren/git_repos/Untitled Folder/optimport.pyx
```

Reply:
This works just fine:

```
# pyimport.py
from scipy.optimize.cython_optimize import zeros
from scipy.optimize.cython_optimize import zeros_struct
from scipy.optimize.cython_optimize import zeros_array
```